### ISSUE TITLE
'python -m pgzero' as used by mu, in a venv.

### ISSUE THREAD

**Issue by lordmauve** [lordmauve/pgzero#63]
Originally reported by: **René Dudfield (Bitbucket: [illume](https://bitbucket.org/illume), GitHub: [illume](https://github.com/illume))**

----------------------------------------

Hello,

I get these errors with virtualenv, and venv with python 3.6 on mac from homebrew. 

Mu issue: https://github.com/mu-editor/mu/issues/349

`python -m pgzero /tmp/lala.py` doesn't work anywhere(in or out of a venv or a virtualenv).
`pgzrun /tmp/lala.py` works without (no venv, or virtualenv).

Versions and such below...

```
20:30:46-rene~$ python3.6 -m venv /tmp/anenv36
20:30:55-rene~$ . /tmp/anenv36/bin/activate
(anenv36) 20:31:03-rene~$ pip install pgzero
Collecting pgzero
Collecting pygame>=1.9 (from pgzero)
  Using cached pygame-1.9.3-cp36-cp36m-macosx_10_9_intel.whl
Installing collected packages: pygame, pgzero
Successfully installed pgzero-1.1 pygame-1.9.3
(anenv36) 20:31:12-rene~$ touch lala.py
(anenv36) 20:31:25-rene~$ python -m pgzero lala.py
2018-02-21 20:31:30.585 Python[83240:10283687] 20:31:30.585 WARNING:  140: This application, or a library it uses, is using the deprecated Carbon Component Manager for hosting Audio Units. Support for this will be removed in a future release. Also, this makes the host incompatible with version 3 audio units. Please transition to the API's in AudioComponent.h.
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pgzero/__main__.py", line 3, in <module>
    main()
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pgzero/runner.py", line 62, in main
    _substitute_full_framework_python()
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pgzero/runner.py", line 55, in _substitute_full_framework_python
    os.execv(framework_python, ['python', '-m', 'pgzero'] + sys.argv[1:])
FileNotFoundError: [Errno 2] No such file or directory
(anenv36) 20:31:30-rene~$ python --version
Python 3.6.4
(anenv36) 20:32:05-rene~$ python -c "import pygame;print(pygame.version.ver);print(pygame.__file__)"
1.9.3
/private/tmp/anenv36/lib/python3.6/site-packages/pygame/__init__.py
(anenv36) 20:32:05-rene~$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.6
BuildVersion:	15G18013
(anenv36) 20:32:05-rene~$ brew --version
Homebrew 1.5.4-48-gc216eff
Homebrew/homebrew-core (git revision 0ec27; last commit 2018-02-20)
(anenv36) 20:32:13-rene~$ which python
/tmp/anenv36/bin/python
(anenv36) 20:32:16-rene~$ python -c "import platform;print(platform.platform())"Darwin-15.6.0-x86_64-i386-64bit
(anenv36) 20:32:51-rene~$ brew search python3
==> Searching local taps...
python3 ✔                                boost-python3
```

```
20:35:20-rene~$ python3 -m virtualenv /tmp/anenv36v
Using base prefix '/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6'
New python executable in /tmp/anenv36v/bin/python3.6
Also creating executable in /tmp/anenv36v/bin/python
Installing setuptools, pip, wheel...done.
20:35:47-rene~$ . /tmp/anenv36v/bin/activate
(anenv36v) 20:35:56-rene~$ pip install pgzero
Collecting pgzero
Collecting pygame>=1.9 (from pgzero)
  Using cached pygame-1.9.3-cp36-cp36m-macosx_10_9_intel.whl
Installing collected packages: pygame, pgzero
Successfully installed pgzero-1.1 pygame-1.9.3
(anenv36v) 20:36:01-rene~$ touch lala.py
(anenv36v) 20:36:05-rene~$ python -m pgzero lala.py
2018-02-21 20:36:22.943 python[83886:10287360] 20:36:22.943 WARNING:  140: This application, or a library it uses, is using the deprecated Carbon Component Manager for hosting Audio Units. Support for this will be removed in a future release. Also, this makes the host incompatible with version 3 audio units. Please transition to the API's in AudioComponent.h.
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/anenv36v/lib/python3.6/site-packages/pgzero/__main__.py", line 3, in <module>
    main()
  File "/tmp/anenv36v/lib/python3.6/site-packages/pgzero/runner.py", line 62, in main
    _substitute_full_framework_python()
  File "/tmp/anenv36v/lib/python3.6/site-packages/pgzero/runner.py", line 55, in _substitute_full_framework_python
    os.execv(framework_python, ['python', '-m', 'pgzero'] + sys.argv[1:])
FileNotFoundError: [Errno 2] No such file or directory
```

----------------------------------------
- Bitbucket: https://bitbucket.org/lordmauve/pgzero/issue/63

**Comment by lordmauve** [comment]
*Original comment by* **Daniel Pope (Bitbucket: [lordmauve](https://bitbucket.org/lordmauve), GitHub: [lordmauve](https://github.com/lordmauve))**:

----------------------------------------

This odd `os.execv()` is the late Rob Collins' fix for #1, which was [pygame #203](https://bitbucket.org/pygame/pygame/issues/203/window-does-not-get-focus-on-os-x-with)

**Comment by lordmauve** [comment]
*Original comment by* **Daniel Pope (Bitbucket: [lordmauve](https://bitbucket.org/lordmauve), GitHub: [lordmauve](https://github.com/lordmauve))**:

----------------------------------------

I pushed a fix in 8c6c06d. Could you please check whether this works for you?

**Comment by lordmauve** [comment]
*Original comment by* **René Dudfield (Bitbucket: [illume](https://bitbucket.org/illume), GitHub: [illume](https://github.com/illume))**:

----------------------------------------

Awesome sauce. After I `pip install numpy` it works.

*happy dance*

Seems there's a dependency on numpy now...
```
00:26:52-rene/tmp$ rm -rf /tmp/anenv36
00:27:07-rene/tmp$ python3.6 -m venv /tmp/anenv36
00:27:12-rene/tmp$ . /tmp/anenv36/bin/activate
(anenv36) 00:27:17-rene/tmp$ pip install hg+http://bitbucket.org/lordmauve/pgzero
Collecting hg+http://bitbucket.org/lordmauve/pgzero
  Cloning hg http://bitbucket.org/lordmauve/pgzero to /private/var/folders/__/l7lzq6ms5g13qpdg59dz1kqc0000gp/T/pip-0g2jyoxg-build
Collecting pygame<2.0,>=1.9.2 (from pgzero==1.2)
  Using cached pygame-1.9.3-cp36-cp36m-macosx_10_9_intel.whl
Installing collected packages: pygame, pgzero
  Running setup.py install for pgzero ... done
Successfully installed pgzero-1.2 pygame-1.9.3
(anenv36) 00:29:40-rene/tmp$ touch lala.py
(anenv36) 00:29:53-rene/tmp$ python -m pgzero lala.py
2018-02-23 00:30:00.959 Python[12096:10940154] 00:30:00.958 WARNING:  140: This application, or a library it uses, is using the deprecated Carbon Component Manager for hosting Audio Units. Support for this will be removed in a future release. Also, this makes the host incompatible with version 3 audio units. Please transition to the API's in AudioComponent.h.
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pgzero/__main__.py", line 1, in <module>
    from pgzero.runner import main
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pgzero/runner.py", line 14, in <module>
    from . import builtins
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pgzero/builtins.py", line 4, in <module>
    from . import tone
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pgzero/tone.py", line 34, in <module>
    import pygame.sndarray
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pygame/sndarray.py", line 47, in <module>
    import pygame._numpysndarray as numpysnd
  File "/private/tmp/anenv36/lib/python3.6/site-packages/pygame/_numpysndarray.py", line 38, in <module>
    import numpy
ModuleNotFoundError: No module named 'numpy'
```
